tests/e2e/p/staking_rewards.go

```diff
@@ -203,18 +226,26 @@ var _ = ginkgo.Describe("[Staking Rewards]", func() {
 				e2e.WithDefaultContext(),
 			)
 			require.NoError(err)
-		})
 
-		deltaDelegatorStartTime := time.Now().Add(e2e.DefaultValidatorStartTimeDiff)
-		tests.Outf("delta delegation period starting at: %v\n", deltaDelegatorStartTime)
+			data, err := pvmClient.GetCurrentValidators(e2e.DefaultContext(), constants.PlatformChainID, []ids.NodeID{alphaNodeID})
+			require.NoError(err)
+			delegatorData := data[0].Delegators[0]
+			effectiveGammaDelegatorDuration = time.Duration(delegatorData.EndTime-delegatorData.StartTime) * time.Second
+		})
```

Comment:
Instead of doing this, we should let `IssueAddPermissionlessDelegatorTx` and every other add staker tx builder the start time picked by the system, along with the txID.
I anticipate this would improve UX and spare some load on our APIs